homeassistant/components/camera/onvif.py

```diff
@@ -67,7 +72,16 @@ def __init__(self, hass, config):
             config.get(CONF_PASSWORD),
             '{}/wsdl/media.wsdl'.format(os.path.dirname(onvif.__file__))
         )
-        self._input = media.GetStreamUri().Uri
+        self._profiles = media.GetProfiles()
+        self._profile_index = config.get(CONF_PROFILE)
+        if self._profile_index >= len(self._profiles):
+            _LOGGER.warning("ONVIF Camera '%s' doesn't provide profile %d."
+                            " Using the last profile.",
+                            self._name, self._profile_index)
+            self._profile_index = -1
+        req = media.create_type('GetStreamUri')
+        req.ProfileToken = self._profiles[self._profile_index]._token
```

Comment:
Access to a protected member is not allowed:
https://travis-ci.org/home-assistant/home-assistant/jobs/328936076#L540

Reply:
And what if that's the only way we can access the token?

Reply:
Make a public function in the onvif-py3 library.

Reply:
onvif-py3 isn't creating this object on it's own. It uses suds library which parses wsdl files. 
Suds is implemented so it adds underscore at the beginning of each attribute that in wsdl has "required" attribute. Please check it here. https://github.com/samkuehn/suds/blob/0fc32c8f418bec47bdfc96224014edb845eaae3e/suds/builder.py#L95
Do you want me to create my own fork of suds? I could also try to change the onvif-py3 to use zeep instead of suds. Maybe it will be better.

Reply:
You should try to add the changes to the official repository

Reply:
As far as I see the official repository is not updated for years, so I guess it won't be that easy

Reply:
It is less than a year, so you should try to make pr: https://github.com/rambo/onvif-py3

Reply:
As I wrote - onvif-py uses suds repository ( https://bitbucket.org/jurko/suds/overview ). 
Suds parses wsdl service response and if response contains attribute that has "required" property it adds "_" prefix to it's name. 
So the fix should be done in suds repository.
I've found a fork of python-onvif - https://github.com/FalkTannhaeuser/python-onvif-zeep it has the same codebase, but uses zeep wsdl parser instead of suds. I've tested it for a while now and seems to work. Maybe I could use it?

Reply:
It's even worse - suds repository doesn't exist anymore.
According to https://pypi.python.org/pypi/suds/0.4 it should be available @ https://fedorahosted.org/suds